components/target_estimator.py

```diff
@@ -0,0 +1,100 @@
+from components.chassis import Chassis
+from components.range_finder import RangeFinder
+from components.turret import Turret
+from components.vision import Vision, VisionData
+
+from utilities.functions import constrain_angle
+
+import math
+import time
+
+
+class TargetEstimator:
+
+    ALPHA = 0.1  # Exponential smoothing fraction
+    CAMERA_TO_LIDAR = 0.15
+
+    chassis: Chassis
+    range_finder: RangeFinder
+    turret: Turret
+    vision: Vision
+
+    def setup(self) -> None:
```

Comment:
```suggestion
    def __init__(self) -> None:
```
since this doesn't require any of the injected components.